docs/contributing/3.0-development/app-layout.rst

```diff
@@ -0,0 +1,241 @@
+features provided by DRF, and features in DRF should be preferred over similar
+features provided by Django.
+
+
+Module Layout
+-------------
+
+This is the basic layout of the ``pulp.app`` package, on the filesystem::
+
+    pulp
+    ├── app
+    │   ├── apps.py
+    │   ├── fields.py
+    │   ├── __init__.py
+    │   ├── logs.py
+    │   ├── management
+    │   │   ├── commands
+    │   │   │   ├── __init__.py
+    │   │   │   └── ...
+    │   │   └── __init__.py
+    │   ├── manage.py
+    │   ├── migrations
+    │   ├── models
+    │   │   ├── __init__.py
+    │   │   └── ...
+    │   ├── pagination.py
+    │   ├── response.py
+    │   ├── serializers
+    │   │   ├── __init__.py
+    │   │   └── ...
+    │   ├── settings.py
+    │   ├── tasks
+    │   │   ├── __init__.py
+    │   │   └── ...
+    │   ├── templates
+    │   │   └── ...
+    │   ├── tests
+    │   │   └── ...
+    │   ├── urls.py
+    │   ├── viewsets
+    │   │   ├── __init__.py
+    │   │   └── ...
+    │   └── wsgi.py
+    └── __init__.py
+
+
+The contents of this package are documented in detail in the :doc:`../platform_api/index`
+documentation. Details how this package is organized can be found
+below, along with information about some of the modules found in this namespace.
+
+.. tip::
+
+    This tree is incomplete, and maybe be out of date. Only the most notable and
+    durable modules have been listed; the most complete listing of modules in this
+    namespace will always be the :doc:`../platform_api/index` documentation.
+
+
+Module Imports
+--------------
+
+For modules in the ``pulp.app`` namespace that are large and capture behaviors
+across multiple concerns of pulp, such as our models, we have broken those
+package up into multiple subpackages. All public identifiers and objects defined
+in submodules are then collected into that module's ``__init__.py``, from which
+they will be imported by other Pulp Platform modules.
+
+Using :mod:`pulp.app.models` as an example, this means that when breaking up the
+``models`` module in ``pulp.app``, the following things are true:
+
+* No models are defined in the ``__init__.py`` of ``pulp.app.models``.
+* All models are defined in submodules located in the ``pulp.app.models`` module
+  directory (where its ``__init__.py`` can be found).
+* The `__init__.py`` in ``pulp.app.models`` should consist only of import statements,
+  ordered to prevent any circular import issues that may result based on the imports
+  that are done in any included submodules.
+* Any models defined in submodules in ``pulp.app.models`` namespace must be imported
+  from the ``pulp.app.models`` namespace, not the submodule in which they are defined.
+  Yes: ``from pulp.app.models import PulpModel``,
+  No: ``from pulp.app.models.pulp import PulpModel``.
+* When adding new models, they must be imported into the ``pulp.app.models``
+  ``__init__.py``, so that they are available to be imported by any other Pulp Platform
```

Comment:
Using "so that" here is gramattically correct, as it indicates a plan of action, not a direct result. The goal of grouping imports in the `__init__.py` is to make things importable in a given namespace. This section isn't explaining *what* happens when you import things this way, it's explaining *why*.